Packages/com.llealloo.audiolink/Runtime/Scripts/YtdlpPlayer.cs

```diff
@@ -0,0 +1,303 @@
+            _reloadURL = EditorGUILayout.Toggle("Reload URL", _reloadURL);
+            if(_reloadURL)
+            {
+                _ytdlpPlayer.UpdateAndPlay();
+                _reloadURL = false;
+            }
+
+            EditorGUI.BeginChangeCheck();
+            EditorGUILayout.LabelField(new GUIContent(" Seek: " + _ytdlpPlayer.PlaybackTimeFormatted(), EditorGUIUtility.IconContent("d_Slider Icon").image));
+            playbackTime = EditorGUILayout.Slider(playbackTime, 0, 1);
+            if(EditorGUI.EndChangeCheck())
+                _ytdlpPlayer.SetPlaybackTime(playbackTime);
+
+            EditorGUI.EndDisabledGroup();
+            
+        }
+    }
+
+    [InitializeOnLoad]
+    public static class YtdlpURLResolver
+    {
+        private static string _ytdlpDownloadURL = "https://github.com/yt-dlp/yt-dlp/releases/latest/download/yt-dlp.exe";
+        private static string _localYtdlpPath = Application.dataPath + "\\AudioLink\\yt-dlp.exe";
+
+        private static string _ytdlpPath = "";
+        private static HashSet<Process> _runningYtdlProcesses = new HashSet<Process>();
+        private static bool _ytdlFound = false;
+
+        /// <summary> Initialize URL Resolver </summary>
+        static YtdlpURLResolver()
+        {
+            EditorApplication.playModeStateChanged += PlayModeChanged;
+            LocateYtdlp();
+        }
+
+        /// <summary> Locate yt-dlp executible, either in VRC application data or locally (offer to download) </summary>
+        public static void LocateYtdlp()
+        {
+            _ytdlFound = false;
+            #if UNITY_EDITOR_WIN
+            string[] splitPath = Application.persistentDataPath.Split('/', '\\');
+            
+            // Check for yt-dlp in VRC application data first
+            _ytdlpPath = string.Join("\\", splitPath.Take(splitPath.Length - 2)) + "\\VRChat\\VRChat\\Tools\\yt-dlp.exe";
+            #elif UNITY_EDITOR_LINUX
+            _ytdlpPath = "/usr/bin/yt-dlp";
+            #endif
+            if (!File.Exists(_ytdlpPath)) 
+            {
+                // Check the local path (in the Assets folder)
+                _ytdlpPath = _localYtdlpPath;
+            }
+
+            if (!File.Exists(_ytdlpPath))
+            {
+                #if UNITY_EDITOR_WIN
+                // Offer to download yt-dlp to the AudioLink folder
+                bool doDownload = EditorUtility.DisplayDialog("[AudioLink] Download yt-dlp?", "AudioLink could not locate yt-dlp in your VRChat folder.\nDownload to AudioLink folder instead?", "Download", "Cancel");
+                if(doDownload)
+                    DownloadYtdlp();
+
+                if(!Application.isPlaying)
+                    EditorApplication.ExitPlaymode();
+                
+                #elif UNITY_EDITOR_LINUX
+                    EditorUtility.DisplayDialog("[AudioLink] Missing yt-dlp", "try \"sudo pacman -S yt-dlp\"", "ok");
```

Comment:
This only works on Arch based distros, and not even the most popular. If you are gonna put a help, at least include a command for Debian.

Reply:
I don't know debian commands and I don't wanna research them; can you help you?

Reply:
`sudo apt install <package>`

Reply:
argh, ytdlp is in a separate PPA, so you'd need to do:
```
sudo add-apt-repository ppa:yt-dlp/stable    # Add ppa repo to apt
sudo apt update                              # Update package list
sudo apt install yt-dlp                      # Install yt-dlp
```

Reply:
So if you wanna keep this, just mention it is specific to Arch

Reply:
Resolved in 631f888e53bf119a63570ed455611d1e38ef151d